Comment out the JANET_NO_UTC_MKTIME defintion. On OpenIndiana with XPG6 the functionality seems to be available.
At least the corresponding test (test/suite-os.janet) is OK.

--- janet-1.37.1/src/include/janet.h.orig	2024-12-06 03:18:16.000000000 +0100
+++ janet-1.37.1/src/include/janet.h	2025-01-12 21:09:03.960283470 +0100
@@ -158,7 +158,7 @@
 
 /* Check sun */
 #ifdef __sun
-#define JANET_NO_UTC_MKTIME
+//#define JANET_NO_UTC_MKTIME
 #endif
 
 /* Define how global janet state is declared */
